### ISSUE TITLE
jsonnet lib generates bogus tenant YAML when mTLS is set

### ISSUE THREAD

**Issue by jpkrohling** [observatorium/observatorium#112]
When a tenant specifies a `mTLS` node with `configMapName` and `caKey`, the resulting `tenants.yaml` contains two entries for the same tenant, one with `.mTLS.caPath` and one with the remaining options. This causes the API to break with the following message:

```console
$ kubectl logs -n observatorium observatorium-xyz-observatorium-api-6d7b475569-jjd82
2020/10/23 14:21:32 failed to parse CA certificate PEM for tenant "test"
```

Here's the full secret that is generated:
```yaml
apiVersion: v1
kind: Secret
metadata:
  labels:
    app.kubernetes.io/component: api
    app.kubernetes.io/instance: observatorium-xyz
    app.kubernetes.io/name: observatorium-api
    app.kubernetes.io/part-of: observatorium
    app.kubernetes.io/version: master-2020-09-08-v0.1.1-142-g61a908f
  name: observatorium-xyz-observatorium-api
  namespace: observatorium
stringData:
  tenants.yaml: |-
    "tenants":
    - "id": "1610b0c3-c509-4592-a256-a1871353dbfa"
      "mTLS":
        "caPath": "/var/run/mtls/test/service-ca.crt"
      "name": "test"
    - "id": "1610b0c3-c509-4592-a256-a1871353dbfa"
      "mTLS":
        "caKey": "service-ca.crt"
        "configMapName": "service-ca-tls"
      "name": "test"
      "oidc":
        "clientID": "test"
        "clientSecret": "ZXhhbXBsZS1hcHAtc2VjcmV0"
        "issuerCAPath": "/var/run/mtls/test/service-ca.crt"
        "issuerURL": "https://dex.dex.svc.cluster.local:5556/dex"
        "usernameClaim": "email"
```

From what I could see, here's the offending code:
https://github.com/observatorium/observatorium/blob/7c57706b8380a63abe4713f92300c29a9a30f335/jsonnet/lib/observatorium-api.libsonnet#L295-L302

Unfortunately, my knowledge of jsonnet is insufficient to fix this issue.

**Comment by squat** [comment]
This is happening because the `mTLS` and `oidc` fields of a tenant's configuration are supposed to be mutually exclusive as they are two different and incompatible authentication strategies. In your example, you specified both mTLS and OIDC settings.  Do yo think it would be better to put an `else ` instead of two `if`s? That way the user error is more obbvious?

**Comment by jpkrohling** [comment]
Sorry, I think I misunderstood what you explained in https://github.com/observatorium/observatorium/pull/110#issuecomment-715328836. Reading it carefully now, I see that you mentioned the existing fields as reference, not that I should be using them. It makes perfect sense now, and I think I can even find my way around with jsonnet now.

**Comment by jpkrohling** [comment]
Closed via #113
